boot/freeldr/freeldr/arch/arm/debug.c

```diff
@@ -0,0 +1,29 @@
+
+/* INCLUDES ***************************************************************/
+
+#include <freeldr.h>
+#include <debug.h>
+
+#define QEMUUART 0x09000000
+
+volatile unsigned int * UART0DR = (unsigned int *) QEMUUART;
```

Comment:
Is that something that works *just* in QEMU?

Reply:
This is for ARM64 and ARM32 QEMU yes, they share the same UART controller. I don't plan on adding support for any specific ARM32 devices UARTs inside freeldr beyond this.